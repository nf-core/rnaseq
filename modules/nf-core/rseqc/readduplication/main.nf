process RSEQC_READDUPLICATION {
    tag "$meta.id"
    label 'process_medium'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/rseqc:5.0.3--py39hf95cd2a_0' :
        'biocontainers/rseqc:5.0.3--py39hf95cd2a_0' }"

    input:
    tuple val(meta), path(bam)

    output:
    tuple val(meta), path("*seq.DupRate.xls"), emit: seq_xls
    tuple val(meta), path("*pos.DupRate.xls"), emit: pos_xls
    tuple val(meta), path("*.pdf")           , emit: pdf
    tuple val(meta), path("*.r")             , emit: rscript
    path  "versions.yml"                     , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    read_duplication.py \\
        -i $bam \\
        -o $prefix \\
        $args

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        rseqc: \$(read_duplication.py --version | sed -e "s/read_duplication.py //g")
    END_VERSIONS
    """
}
