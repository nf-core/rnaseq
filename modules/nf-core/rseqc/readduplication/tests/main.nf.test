nextflow_process {

    name "Test Process RSEQC_READDUPLICATION"
    script "../main.nf"
    process "RSEQC_READDUPLICATION"
    tag "modules"
    tag "modules_nfcore"
    tag "rseqc"
    tag "rseqc/readduplication"

    test("sarscov2 paired-end [bam]") {

        when {
            process {
                """
                input[0] = [ [ id:'test', single_end: false ], // meta map
                             file(params.test_data['sarscov2']['illumina']['test_paired_end_sorted_bam'], checkIfExists: true)
                           ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.seq_xls).match("seq_xls") },
                { assert snapshot(process.out.pos_xls).match("pos_xls") },
                { assert snapshot(process.out.rscript).match("rscript") },
                { assert snapshot(process.out.versions).match("versions") },
                { assert snapshot(file(process.out.pdf[0][1]).name).match("pdf") }
            )
        }

    }

}
