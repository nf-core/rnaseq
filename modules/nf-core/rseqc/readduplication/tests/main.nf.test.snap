{
    "pos_xls": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.pos.DupRate.xls:md5,a859bc2031d46bf1cc4336205847caa3"
                ]
            ]
        ],
        "timestamp": "2023-11-24T00:52:45.27809706"
    },
    "rscript": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.DupRate_plot.r:md5,3c0325095cee4835b921e57d61c23dca"
                ]
            ]
        ],
        "timestamp": "2023-11-24T00:52:45.285457599"
    },
    "pdf": {
        "content": [
            "test.DupRate_plot.pdf"
        ],
        "timestamp": "2023-11-24T00:52:45.304152477"
    },
    "seq_xls": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.seq.DupRate.xls:md5,ee8783399eec5a18522a6f08bece338b"
                ]
            ]
        ],
        "timestamp": "2023-11-24T00:52:45.262656562"
    },
    "versions": {
        "content": [
            [
                "versions.yml:md5,8a0721f3247f97135eadb8eecbe142a1"
            ]
        ],
        "timestamp": "2023-12-14T17:23:41.82098803"
    }
}